core/server/services/settings/index.js

```diff
@@ -4,16 +4,78 @@
  */
 
 var SettingsModel = require('../../models/settings').Settings,
-    SettingsCache = require('./cache');
+    SettingsCache = require('./cache'),
+    SettingsLoader = require('./loader'),
+    _ = require('lodash'),
+    common = require('../../lib/common');
+    // SettingsFiles = require('./ensure-settings');
 
 module.exports = {
     init: function init() {
+        // TODO: the ensure settings fn up before the settings cache
+        // return SettingsFiles(this.knownSettings()).then(() => {
+
         // Update the defaults
         return SettingsModel.populateDefaults()
             .then(function (settingsCollection) {
                 // Initialise the cache with the result
                 // This will bind to events for further updates
                 SettingsCache.init(settingsCollection);
             });
+        // });
+    },
+
+    // TODO: is this the right place for it? We need a global place where
+    // we can add additional files that are supported.
+    // This can also be a config value for example.
+    knownSettings: function knownSettings() {
+        return ['routes'];
+    },
+
+    /**
+     * Getter for YAML settings objects.
+     * Example: `settingsSrvc.get('routes').then(...)`
+     * @param {String} setting type of setting. Needs to be supported.
+     *                         see `knownSettings`
+     * @returns {Promise<Object>} settingsFile
+     * @description Returns settings object as defined per YAML files in
+     * `/content/settings` directory.
+     */
+    get: function get(setting) {
+        const knownSettings = this.knownSettings();
+
+        if (!setting || _.indexOf(knownSettings, setting) < 0) {
+            return Promise.reject(new common.errors.GhostError({
+                message: `Requested setting is not available: '${setting}'.`,
+                help: `Please use only the supported settings: ${knownSettings}.`
+            }));
+        }
+
+        return SettingsLoader()
+            .then((settingsFile) => {
+                if (_.isEmpty(settingsFile) || !settingsFile[setting]) {
```

Comment:
Can you explain the use cases and your think here a bit further? 

The routes file has always a fallback, or am i wrong?
And i guess any upcoming files will have a fallback as well. Furthermore, this is internal logic.

If there is a case, we have to throw an error, then i expect that the settings loader throws an error.

Reply:
> The routes file has always a fallback, or am i wrong?

I doesn't really have a fallback at the moment. The fallback is, that we have to make sure that the file exists before we get the routes. Do you think we should add a real fallback in case we can't access the file in the directory?

Yes, it is internal logic and would only be caused when internally used wrong.

Reply:
> Do you think we should add a real fallback in case we can't access the file in the directory?

If you can't read `content/settings/routes.yml`, this is a permission problem of the installation. Ghost should throw this error.

I am not sure what this condition is for

> if (_.isEmpty(settingsFile) || !settingsFile[setting]) {

Reply:
I decided to remove this part, as it really shouldn't happen. The only case, I can think of is a permission problem as well. I will make sure, that the tests will cover this behaviour.